src/Database/Bloodhound/Types.hs

```diff
@@ -604,7 +606,16 @@ data Search = Search { queryBody       :: Maybe Query
                        -- default False
                      , trackSortScores :: TrackSortScores
                      , from            :: From
-                     , size            :: Size } deriving (Eq, Show)
+                     , size            :: Size 
+                     , searchType      :: SearchType } deriving (Eq, Show)
+
+data SearchType = SearchTypeQueryThenFetch 
+                | SearchTypeDfsQueryThenFetch
+                | SearchTypeCount
+                | SearchTypeScan
+                | SearchTypeQueryAndFetch
+                | SearchTypeDfsQueryAndFetch
+  deriving (Eq, Show)
```

Comment:
This is reasonable and I won't block on this, but I get a bit suspicious when a product-of-sum and maybe fields are combined. Usually means you want to lift the sum up a level and case out when fields will or will not occur in the variants.